### PR TITLE
New resource: aws_lightsail_disk

### PR DESCRIPTION
**Pull request by blckct**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

Add new resource: aws_lightsail_disk

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
aws/lightsail_disk: New resource
```

Output from acceptance testing:

```
make testacc TEST=./aws TESTARGS='-run=TestAccAWSLightsailDisk_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -parallel 20 -run=TestAccAWSLightsailDisk_basic -timeout 120m
=== RUN   TestAccAWSLightsailDisk_basic
=== PAUSE TestAccAWSLightsailDisk_basic
=== CONT  TestAccAWSLightsailDisk_basic
--- PASS: TestAccAWSLightsailDisk_basic (30.49s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	30.519s

make testacc TEST=./aws TESTARGS='-run=TestAccAWSLightsailDisk_Tags'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -parallel 20 -run=TestAccAWSLightsailDisk_Tags -timeout 120m
=== RUN   TestAccAWSLightsailDisk_Tags
=== PAUSE TestAccAWSLightsailDisk_Tags
=== CONT  TestAccAWSLightsailDisk_Tags
--- PASS: TestAccAWSLightsailDisk_Tags (52.15s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	52.177s
...
```

### CODE REVIEW

**Comment by missinglink** [comment]
Hi @blckct, this is super cool, thanks for opening it.

Do you possible also have code which performs the disk attachment to a running or stopped lightsail instance as per https://github.com/aws/aws-sdk-go/blob/master/service/lightsail/api.go#L20 ?

**Comment by blckct** [comment]
@missinglink Hey, 
I'm afraid I don't have the code for attachment (at least not completed). I wanted to get feedback on this one first before I proceed any further.

File: website/docs/r/lightsail_disk.html.markdown [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+Provides a Lightsail Disk. Amazon Lightsail is a service to provide easy virtual private servers
+with custom software already setup. See [What is Amazon Lightsail?](https://lightsail.aws.amazon.com/ls/docs/getting-started/article/what-is-amazon-lightsail)
+for more information.
+
+~> **Note:** Lightsail is currently only supported in a limited number of AWS Regions, please see ["Regions and Availability Zones in Amazon Lightsail"](https://lightsail.aws.amazon.com/ls/docs/overview/article/understanding-regions-and-availability-zones-in-amazon-lightsail) for more details
+
+## Example Usage
+
+```hcl
+# Create a new GitLab Lightsail Instance
```
Comment (missinglink) [inline website/docs/r/lightsail_disk.html.markdown]: This comment refers to `instance` rather than `disk`

Comment (blckct) [reply website/docs/r/lightsail_disk.html.markdown]: You're right. Fixed.

**Comment by blckct** [comment]
Rebased onto current master.

**Comment by teamterraform** [comment]
## Notification of Recent and Upcoming Changes to Contributions

Thank you for this contribution! There have been a few recent development changes that affect this pull request. We apologize for the inconvenience, especially if there have been long review delays up until now. Please note that this is automated message from an unmonitored account. See the [FAQ](https://github.com/terraform-providers/terraform-provider-aws/blob/master/docs/FAQ.md) for additional information on the maintainer team and review prioritization.

If you are unable to complete these updates, please leave a comment for the community and maintainers so someone can potentially continue the work. The maintainers will encourage other contributors to use the existing contribution as the base for additional changes as appropriate. Otherwise, contributions that do not receive updated code or comments from the original contributor may be closed in the future so the maintainers can focus on active items.

For the most up to date information about Terraform AWS Provider development, see the [Contributing Guide](https://github.com/terraform-providers/terraform-provider-aws/blob/master/docs/CONTRIBUTING.md). Additional technical debt changes can be tracked with the [`technical-debt` label on issues](https://github.com/terraform-providers/terraform-provider-aws/issues?q=is%3Aopen+is%3Aissue+label%3Atechnical-debt).

As part of updating a pull request with these changes, the most current unit testing and linting will run. These may report issues that were not previously reported.

### Action Required: Terraform 0.12 Syntax

Reference: https://github.com/terraform-providers/terraform-provider-aws/issues/8950
Reference: https://github.com/terraform-providers/terraform-provider-aws/issues/14417

Version 3 and later of the Terraform AWS Provider, which all existing contributions would potentially be added, only supports Terraform 0.12 and later. Certain syntax elements of Terraform 0.11 and earlier show deprecation warnings during runs with Terraform 0.12. Documentation and test configurations, such as those including deprecated string interpolations (`some_attribute = "${aws_service_thing.example.id}"`) should be updated to the newer syntax (`some_attribute = aws_service_thing.example.id`). Contribution testing will automatically fail on older syntax in the near future. Please see the referenced issues for additional information.

### Action Required: Terraform Plugin SDK Version 2

Reference: https://github.com/terraform-providers/terraform-provider-aws/issues/14551

The Terraform AWS Provider has been upgraded to the latest version of the Terraform Plugin SDK. Generally, most changes to contributions should only involve updating Go import paths in source code files. Please see the referenced issue for additional information.

### Action Required: Removal of website/aws.erb File

Reference: https://github.com/terraform-providers/terraform-provider-aws/issues/14712

Any changes to the `website/aws.erb` file are no longer necessary and should be removed from this contribution to prevent merge issues in the near future when the file is removed from the repository. Please see the referenced issue for additional information.

### Upcoming Change of Git Branch Naming

Reference: https://github.com/terraform-providers/terraform-provider-aws/issues/14292

Development environments will need their upstream Git branch updated from `master` to `main` in the near future. Please see the referenced issue for additional information and scheduling.

### Upcoming Change of GitHub Organization

Reference: https://github.com/terraform-providers/terraform-provider-aws/issues/14715

This repository will be migrating from https://github.com/terraform-providers/terraform-provider-aws to https://github.com/hashicorp/terraform-provider-aws. No practitioner or developer action is anticipated and most GitHub functionality will automatically redirect to the new location. Go import paths including `terraform-providers` can remain for now. Please see the referenced issue for additional information and scheduling.

**Comment by zhelding** [comment]
Hi @blckct. No further action is required on your end for merging this pull request.

This pull request will now be addressed via our standard prioritization process. Unfortunately, given the large quantity of issues and pull requests we receive, we are not always able to review every pull request in a timely manner.

We prioritize contributions that provide the greatest value to the largest number of users -- which we primarily infer based on the number of :thumbsup: [reactions](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/).  Additionally, an explanation of the particular use case for a contribution can provide information on how the changes might be useful in a broader context.

You can get a view of the wider development plans for the provider via our [road map](https://github.com/hashicorp/terraform-provider-aws/blob/main/ROADMAP.md), published quarterly.

Thank you again for your contribution and your patience.

**Comment by rpolnx** [comment]
Any updates about this PR?

**Comment by blckct** [comment]
@rpolnx It seems like it's been deprioritized for a long time like all the other Lightsail changes. I saw someone created a separate provider over here: https://github.com/DeYoungTech/terraform-provider-awslightsail but I haven't tested it.

**Pull request closed without merging**